lib/localisation/src/preprocessing/accelerometer.rs

```diff
@@ -1 +1,300 @@
-//test
+use super::super::types::{K_NUM_ACCELEROMETERS, K_NUM_ALLOWED_ACCELEROMETER_OUTLIERS, K_NUM_AXIS};
+use heapless::Vec;
+use hyped_core::types::{AccelerometerData, RawAccelerometerData, SensorChecks};
+
+#[allow(dead_code)]
+pub struct Quartiles {
+    q1: f32,
+    q2: f32,
+    q3: f32,
+
+    iqr: f32,
+    lower_bound: f32,
+    upper_bound: f32,
+}
+
+impl Quartiles {
+    pub fn new(q1: f32, q2: f32, q3: f32, is_unreliable: bool) -> Self {
+        let mut bound_factor = 1.5;
+        if is_unreliable {
+            bound_factor = 1.2;
+        }
+
+        Self {
+            q1,
+            q2,
+            q3,
+            iqr: q3 - q1,
+            lower_bound: q1 - bound_factor * (q3 - q1),
+            upper_bound: q3 + bound_factor * (q3 - q1),
+        }
+    }
+}
+
+pub struct AccelerometerPreprocessor {
+    num_reliable_accelerometers: i32,
+    reliable_accelerometers: [bool; K_NUM_ACCELEROMETERS],
+    num_outliers_per_accelerometer: [i32; K_NUM_ACCELEROMETERS],
+}
+
+impl Default for AccelerometerPreprocessor {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl AccelerometerPreprocessor {
+    pub fn new() -> Self {
+        Self {
+            num_reliable_accelerometers: K_NUM_ACCELEROMETERS as i32,
+            reliable_accelerometers: [true; K_NUM_ACCELEROMETERS],
+            num_outliers_per_accelerometer: [0; K_NUM_ACCELEROMETERS],
+        }
+    }
+
+    pub fn handle_outliers(
+        &mut self,
+        data: AccelerometerData<K_NUM_ACCELEROMETERS>,
+    ) -> Option<AccelerometerData<K_NUM_ACCELEROMETERS>> {
+        let quartiles = match self.calculate_quartiles(&data) {
+            Some(quartiles) => quartiles,
+            None => return None,
+        };
```

Comment:
```rust
let quartiles = self.calculate_quartiles(&data)?;
```
`?` operator will give back the wrapped value of a `Some(...)` or early return if `None`